### PR TITLE
feat(ui-results): add `MatrixGridSynthesis`

### PR DESCRIPTION
**Pull request by hdinia**
ANT-2416

### CODE REVIEW

File: webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ import ResultFilters from "./ResultFilters.tsx";
 import { toError } from "../../../../../../utils/fnUtils.ts";
 import EmptyView from "../../../../../common/page/SimpleContent.tsx";
 import { getStudyMatrixIndex } from "../../../../../../services/api/matrix.ts";
+import { MatrixGridSynthesis } from "@/components/common/Matrix/components/MatrixGridSynthesis/index.tsx";
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx:68]: It's normal `index.tsx` at the end?

File: webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import { useMemo } from "react";
+import { darkTheme, readOnlyDarkTheme } from "../../styles";
+import { formatGridNumber } from "../../shared/utils";
+
+type CellValue = number | string;
+
+interface MatrixGridSynthesisProps {
+  data: CellValue[][];
+  columns: GridColumn[];
+  width?: string;
```
Comment (skamril) [inline webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx:31]: You can allow number:

```typescript
width?: string | number;
height?: string | number;
```

File: webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        return {
+          kind: GridCellKind.Number,
+          data: value,
+          displayData: formatGridNumber({ value, maxDecimals: 3 }),
+          decimalSeparator: ".",
+          thousandSeparator: " ",
+          readonly: true,
+          allowOverlay: false,
+          contentAlign: "right",
+        } satisfies NumberCell;
```
Comment (skamril) [inline webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx:65]: I didn't know about `satisfies`, nice

File: webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    },
+    [data],
+  );
+
+  ////////////////////////////////////////////////////////////////
+  // JSX
+  ////////////////////////////////////////////////////////////////
+
+  return (
+    <div style={{ width, height }}>
```
Comment (skamril) [inline webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx:84]: To follow my previous comment use `<Box>` with `sx`

File: webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx (hunk 1 above)
Comment (hdinia) [reply webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx:68]: hum it should not appear.. dont know why the auto import added this I will check it, thanks

File: webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx (hunk 2 above)
Comment (hdinia) [reply webapp/src/components/common/Matrix/components/MatrixGridSynthesis/index.tsx:31]: true but we usually don't modify this, I will add it tho

**Comment by hdinia** [comment]
moved to: https://github.com/AntaresSimulatorTeam/AntaREST/pull/2233

**Pull request closed without merging**